Build from source fails because of libjpeg_turbo.cmake on some Linux based OS

I do not manage to build the current master from source because of an issue with `libjpeg_turbo.cmake`. In order to reproduce the error you would have to clone the repo, create a build folder, go to that folder and call `cmake .. & make`.

My system is Fedora 27, but I assume it is an issue with Linux in general, I assume it might work on other platforms.

When I build the current master, `make` crashes with the following error message:

```
make[2]: *** No rule to make target '3rdparty_install/lib/libturbojpeg.a', needed by 'bin/ConvertPointCloud'.  Stop.
make[1]: *** [CMakeFiles/Makefile2:1309: src/Tools/CMakeFiles/ConvertPointCloud.dir/all] Error 2
```

I managed to compile `Open3D` by changing line 70 in `3rdparty/libjpeg-turbo/libjpeg-turbo.cmake` from
```
    ${3RDPARTY_INSTALL_PREFIX}/lib/${CMAKE_STATIC_LIBRARY_PREFIX}${lib_name}${CMAKE_STATIC_LIBRARY_SUFFIX}
```
to
```
${3RDPARTY_INSTALL_PREFIX}/lib64/${CMAKE_STATIC_LIBRARY_PREFIX}${lib_name}${CMAKE_STATIC_LIBRARY_SUFFIX}
```
(change of `lib` to `lib64`).

I assume that this would break the build on other system such as windows.

--- a/3rdparty/libjpeg-turbo/libjpeg-turbo.cmake
+++ b/3rdparty/libjpeg-turbo/libjpeg-turbo.cmake
@@ -1,5 +1,12 @@
 include(ExternalProject)
 
+if (WIN32)
+    set(LIBDIR "lib")
+else()
+    include(GNUInstallDirs)
+    set(LIBDIR ${CMAKE_INSTALL_LIBDIR})
+endif()
+
 # Set compiler flags
 if ("${CMAKE_C_COMPILER_ID}" STREQUAL "MSVC")
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} /D_CRT_SECURE_NO_WARNINGS")
@@ -67,7 +74,7 @@ endif()
 set(JPEG_TURBO_LIBRARIES ${lib_name})
 
 set(turbojpeg_LIB_FILES
-    ${3RDPARTY_INSTALL_PREFIX}/lib/${CMAKE_STATIC_LIBRARY_PREFIX}${lib_name}${CMAKE_STATIC_LIBRARY_SUFFIX}
+    ${3RDPARTY_INSTALL_PREFIX}/${LIBDIR}/${CMAKE_STATIC_LIBRARY_PREFIX}${lib_name}${CMAKE_STATIC_LIBRARY_SUFFIX}
 )
 
 target_include_directories(turbojpeg SYSTEM INTERFACE

